storm-client/src/jvm/org/apache/storm/executor/spout/SpoutOutputCollectorImpl.java

```diff
@@ -22,19 +22,23 @@
 import org.apache.storm.executor.TupleInfo;
 import org.apache.storm.spout.ISpout;
 import org.apache.storm.spout.ISpoutOutputCollector;
+import org.apache.storm.tuple.AddressedTuple;
 import org.apache.storm.tuple.MessageId;
 import org.apache.storm.tuple.TupleImpl;
 import org.apache.storm.tuple.Values;
-import org.apache.storm.utils.Utils;
 import org.apache.storm.utils.MutableLong;
 import org.apache.storm.utils.RotatingMap;
+import org.apache.storm.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Random;
 
+// Methods are not thread safe. Each thread expected to have a separate instance, or else synchronize externally
```

Comment:
nit: better to make it as javadoc so that it can be exposed to more ways.

As @revans2 stated, I also think this is a good assumption for a spout, but even better to update the restriction if we have documented any. That's just a 2 cents, not blocker.

Reply:
To nail down and document the concurrent emits semantics I had opened [STORM-2945](https://issues.apache.org/jira/browse/STORM-2945)

Reply:
I thought STORM-2945 was filed to find the way to support background emit without external synchronization, so likely having the chance to keep unresolved in 2.x. If you intended to document how to enable background emit with current state in STORM-2945, please ignore the comment here.

Reply:
It is to figure out what will have for Storm 2.0... since we cannot make any breaking changes even if we like to thereafter until 3.0.

Reply:
OK. Please let me know if you plan to figure out in time frame of Storm 2.0.0. I'll add it in epic of releasing Storm 2.0.0. Thanks!

Reply:
yes please thats the intent.